OpenTabletDriver.UX/MainForm.cs

```diff
@@ -530,8 +540,34 @@ private async Task SaveSettings()
             }
         }
 
+        private CancellationTokenSource _disableApplySaveButtons = new();
+
+        private void DisableApplySaveButtons(bool disableSave = true)
+        {
+            Application.Instance.InvokeAsync(async () =>
+            {
+                await _disableApplySaveButtons.CancelAsync();
+                _disableApplySaveButtons = new CancellationTokenSource();
+                if (disableSave)
+                    saveButton.Enabled = false;
+
+                applyButton.Enabled = false;
+
+                // debounce re-enabling task
+                await Task.Delay(2500, _disableApplySaveButtons.Token).ContinueWith(task =>
+                {
+                    if (!task.IsCompletedSuccessfully) return;
```

Comment:
Should this be:
```suggestion
                    if (task.IsCanceled) return;
```

If for some insane reason this were to get `IsFaulted` (due to an error) it seems sensible to have the failstate be re-enabling the buttons.

The three states here being `IsCompletedSuccessfully`, `IsFaulted`, and `IsCanceled`.